codebuild/spec/buildspec_sidetrail.yml

```diff
@@ -0,0 +1,44 @@
+---
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You may not use
+# this file except in compliance with the License. A copy of the License is
+# located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing permissions and
+# limitations under the License.
+version: 0.2
+
+phases:
+  install:
+    runtime-versions:
+      python: 3.7
+    commands:
+      - echo Entered the install phase...
+      - apt-key list
+      - apt-key adv --keyserver hkp://keyserver.ubuntu.com:80 --recv-keys 6B05F25D762E3157
+      - add-apt-repository ppa:ubuntu-toolchain-r/test -y
+      - apt-get update -o Acquire::CompressionTypes::Order::=gz
+      - apt-get update -y
+      - |
+        if expr "${GCC_VERSION}" : "9" >/dev/null; then
+          apt-get install -y --no-install-recommends g++ g++-9 gcc gcc-9;
+        fi
+      - |
+        if expr "${GCC_VERSION}" : "6" >/dev/null; then
+          apt-get install -y --no-install-recommends g++ g++-6 gcc gcc-6;
+        fi
+      - apt-get install -y --no-install-recommends indent kwstyle lcov libssl-dev clang-3.9 llvm-3.9 m4 make net-tools nettle-bin nettle-dev pkg-config psmisc python3-pip shellcheck sudo tcpdump unzip valgrind zlib1g-dev zlibc
+  build:
+    commands:
+      - printenv
+      - codebuild/bin/run_sidetrail.sh /sidetrail-install-dir ${CODEBUILD_SRC_DIR}
```

Comment:
Is this the only diff between this file and the other buildspec file?  I could go either way on buildspec vs bash, but we should be consistent- if we create multiple buildspecs, the `s2n_codebuild.sh` script should get cleaned up, as it handles most of the branching logic for tests.

Reply:
I removed sidetrail from `s2n_codebuild.sh`. If I understand the sidetrail tests correctly, we don't even need to build S2N. We are just building specific files and running tests against those files. This will help keep the time down on the sidetrail tests.

I wouldn't mind sticking with buildspec files, but I don't know how to weigh the benefits between the options. In the future, I would like one test to verify comments/copyrights/etc, another does the unit tests, other tests would only build what they need (I'm thinking about the fuzzers here). I'm not sure if spec files get us closer to that future or not.

If we can remove the redundant testing we could speed up the entire build process.